pkg/tracking/store/sql/runs.go

```diff
@@ -251,61 +246,167 @@ func applyFilter(logger *logrus.Logger, database, transaction *gorm.DB, filter s
 	return nil
 }
 
+type orderByExpr struct {
+	identifier *string
+	key        string
+	order      *string
+}
+
+var ErrInvalidOrderClauseInput = errors.New("input string is empty or only contains quote characters")
+
+const (
+	tableAndColumn = 2
+	startTime      = "start_time"
+	name           = "name"
+)
+
+func orderByKeyAlias(input string) string {
+	switch input {
+	case "created", "Created":
+		return startTime
+	case "run_name", "run name", "Run name", "Run Name":
+		return name
+	case "run_id":
+		return "run_uuid"
+	default:
+		return input
+	}
+}
+
+func handleInsideQuote(
+	char, quoteChar rune, insideQuote *bool, current *strings.Builder, result *[]string,
+) {
+	if char == quoteChar {
+		*insideQuote = false
+
+		*result = append(*result, current.String())
+		current.Reset()
+	} else {
+		current.WriteRune(char)
+	}
+}
+
+func handleOutsideQuote(char rune, insideQuote *bool, quoteChar *rune, current *strings.Builder, result *[]string) {
+	switch char {
+	case ' ':
+		if current.Len() > 0 {
+			*result = append(*result, current.String())
+			current.Reset()
+		}
+	case '"', '\'', '`':
+		*insideQuote = true
+		*quoteChar = char
+	default:
+		current.WriteRune(char)
+	}
+}
+
+func splitWithQuotes(input string) []string {
```

Comment:
This is an implementation detail covered by the current tests. Having additional tests for these functions feels too tightly coupled for my taste.

One other readability issue here is that these are separate functions because the cyclomatic linter started complaining.